### ISSUE TITLE
why does WLAN need VLAN tagging, or why doesn't RPI support that?

### ISSUE THREAD

**Issue by mcr** [nqminds/EDGESec#80]
When running edgesec on an RPI4, I am getting:

    2021-12-16 19:09:42.843  TRACE iw.c:199: phy0 -> P2P-device
    2021-12-16 19:09:42.843  TRACE iw.c:377: WiFi interface wlan0 doesn't suport vlan tagging
    2021-12-16 19:09:42.843  DEBUG engine.c:338: interface wlan0 not VLAN capable
    2021-12-16 19:09:42.843  TRACE generic_hsm_driver.c:69: context param is NULL
    Failed to start edgesec engine.

This is with mcr@fennrau:~/EDGESec/build $ uname -a
Linux fennrau 4.19.97-v7l+ #1294 SMP Thu Jan 30 13:21:14 GMT 2020 armv7l GNU/Linux

Perhaps I am running too old a kernel.

**Comment by mereacre** [comment]
The ```hostapd``` daemon, which has the role of creating the software WiFi AP, needs the ```AP/VLAN``` interface mode of a WIFI card/modem to map (see the option ```vlan_file``` from [https://w1.fi/cgit/hostap/plain/hostapd/hostapd.conf](https://w1.fi/cgit/hostap/plain/hostapd/hostapd.conf])) from VLAN IDs to virtual network interfaces that correspond to a give subnet.

```edgesec``` has the option ([https://github.com/nqminds/EDGESec/blob/main/config.ini.in#L11](https://github.com/nqminds/EDGESec/blob/main/config.ini.in#L11)) to detect if a WIFI card supports the ```AP/VLAN``` interface mode. You can check for yourself by running the following command:
```bash
iw phy
```
and then subsequently check the ```Supported interface modes``` category.

The ```edgesec``` tool was tested with several WIFI cards. More details in [https://nqminds.github.io/EDGESec/docs/devices](https://nqminds.github.io/EDGESec/docs/devices).

For Raspberry Pis the card that works out of the box is this USB one [https://thepihut.com/products/usb-wifi-adapter-for-the-raspberry-pi](https://thepihut.com/products/usb-wifi-adapter-for-the-raspberry-pi). It has the following USB options (```lsusb```):
```
148f:5370 Ralink Technology, Corp. RT5370 Wireless Adapter
```

**Comment by aloisklink** [comment]
Just to add some more information in layman terms (since other people might see this same issue):

VLAN tagging is used to segment the network. E.g. if you have a bunch of smart light-bulbs from "Brand-Example", VLAN tagging can be used to put them all on the same subnet, to prevent them from accessing other devices on your network.

The Raspberry Pi 4's internal WiFi doesn't yet support VLAN tagging, however, the most common USB wifi adaptor for Raspberry Pi's does support it: https://thepihut.com/products/usb-wifi-adapter-for-the-raspberry-pi.

This would normally be called `wlan1` instead of `wlan0` on a Raspberry Pi.

**Comment by mcr** [comment]
> The `hostapd` daemon, which has the role of creating the software WIFI access point (AP), needs the `AP/VLAN` interface mode of a WIFI card/modem to map (see the option `vlan_file` from [https://w1.fi/cgit/hostap/plain/hostapd/hostapd.conf](https://w1.fi/cgit/hostap/plain/hostapd/hostapd.conf%5D)) from VLAN IDs to virtual network interfaces that correspond to given subnets.

Why do we need to do this?
Why do we need more than one ESSID? or more than one subnet?
OpenWRT will all of this itself.
